docs/ideas/yaml/jmespath.toml

```diff
@@ -0,0 +1,89 @@
+# The values below were taken from .github/workflows/python.yaml in this repo
+
+# 1. JMESPath as part of the section name, after the file name.
+# Everything after the file name is considered a JMESPath https://jmespath.org/
+# Format: ["path/to/file.ext".jmes.path.expression]
+#
+# 2. "jobs.build.strategy.matrix" should have "os" and "python-version"
+# 3. Both are lists, and they have to be exactly as described here.
+[".github/workflows/python.yaml".jobs.build.strategy.matrix]
+os = ["ubuntu-latest", "macos-latest", "windows-latest"]
+"python-version" = ["3.6", "3.7", "3.8", "3.9"]
+
+# 4. "jobs.build" should have "runs-on" with value "${{ matrix.os }}"
+[".github/workflows/python.yaml".jobs.build]
+"runs-on" = "${{ matrix.os }}"
+
+# 5. "{{" and "}}" will conflict with Jinja https://github.com/andreoliwa/nitpick/issues/283
+# So we need a way to turn on/off Jinja templating.
+# Probably "false" will be the default, to keep compatibility.
+# Whoever wants to use Jinja will need to set "true" either here or as a global config on .nitpick.toml
+__jinja = false
+
+# 6. Another way to turn off Jinja for a specific key only, not the whole dict
+# (using the "__" syntax from Django filters, SQLAlchemy, factoryboy...)
+"runs-on__no_jinja" = "${{ matrix.os }}"
+
+# 7. Same effect as items 1, 2 and 3, but with a different syntax.
+# 8. Everything that is not a dunder key ("__") will be a dict to be enforced.
+[[".github/workflows/python.yaml".contains]]
+__jmespath = "jobs.build.strategy.matrix"
+os = ["ubuntu-latest", "macos-latest", "windows-latest"]
+"python-version" = ["3.6", "3.7", "3.8", "3.9"]
+
+# 9. Same as item 4, but with a different syntax.
+[[".github/workflows/python.yaml".contains]]
+__jmespath = "jobs.build"
+"runs-on" = "${{ matrix.os }}"
+
+# 10. "jobs.build.steps" can have multiple dicts; each one is a "contains" table
+[[".github/workflows/python.yaml".contains]]
+__jmespath = "jobs.build.steps"
+uses = "actions/checkout@v2"
+
+# 11. The "problem" with the "contains" table is that "__jmespath" will be repeated many times
```

Comment:
As you are saying, in this case this approach now seems a bit bloated.

Going back to the first approach would be something like:
```
[[".github/workflows/python.yaml".jobs.build.steps]]
uses = "actions/checkout@v2"

[[".github/workflows/python.yaml".jobs.build.steps]]
name = "Set up Python ${{ matrix.python-version }}"
uses = "actions/setup-python@v2"
with = {"python-version" = "${{ matrix.python-version }}"}
```
Neat.

Reply:
Hmm... that is something.
And it seems to work, it can be parsed when flattened:

- [neat.json](https://gist.github.com/andreoliwa/0dbe523468f421b27901ed28d89485dd#file-neat-json)
- [neat-flattened.json](https://gist.github.com/andreoliwa/0dbe523468f421b27901ed28d89485dd#file-neat-flattened-json)

I think we have a lead here. 🔍 
This should be the first approach to be tried on the actual implementation.